apps/dashboard/src/components/engine/overview/import-backend-wallet-button.tsx

```diff
@@ -10,204 +9,327 @@ import {
+                          form.getFieldState("label", form.formState).error
+                            ?.message
+                        }
+                        htmlFor="wallet-label"
+                        isRequired={false}
+                        tooltip={null}
+                      >
+                        <Input
+                          id="wallet-label"
+                          type="text"
+                          placeholder="A description to identify this backend wallet"
+                          {...form.register("label")}
+                        />
+                      </FormFieldSetup>
 
-              {walletConfig?.type === "aws-kms" && (
-                <div className="flex flex-col gap-4">
-                  <FormFieldSetup
-                    label="AWS KMS Key ID"
-                    isRequired
-                    errorMessage={
-                      form.getFieldState("awsKmsKeyId", form.formState).error
-                        ?.message
-                    }
-                    htmlFor="awsKmsKeyId"
-                  >
-                    <Input
-                      id="awsKmsKeyId"
-                      placeholder=""
-                      autoComplete="off"
-                      type="text"
-                      {...form.register("awsKmsKeyId", { required: true })}
-                    />
-                  </FormFieldSetup>
+                      {/* Local */}
+                      {walletType === "local" ? (
+                        <FormFieldSetup
+                          key="privateKey"
+                          htmlFor="privateKey"
+                          label="Private key"
+                          errorMessage={
+                            form.getFieldState("privateKey", form.formState)
+                              .error?.message
+                          }
+                          isRequired
+                          tooltip={null}
+                        >
+                          <Input
+                            id="privateKey"
+                            placeholder="Your wallet private key"
+                            autoComplete="off"
+                            type="text"
+                            {...form.register("privateKey", {
+                              required: true,
+                              // TODO: add private key validation here
+                            })}
+                          />
+                        </FormFieldSetup>
+                      ) : walletType === "aws-kms" ? (
+                        <FormFieldSetup
+                          isRequired
+                          label="AWS KMS ARN"
+                          errorMessage={
+                            form.getFieldState("awsKmsArn", form.formState)
+                              .error?.message
+                          }
+                          htmlFor="awsKmsArn"
+                          tooltip={null}
+                        >
```

Comment:
add `key` to all the `FormFieldSetup` components here to avoid having wrong values in the inputs